typhos/display.py

```diff
@@ -34,6 +36,15 @@ class DisplayTypes(enum.IntEnum):
     detailed_screen = 1
     engineering_screen = 2
 
+    @property
+    def friendly_name(self) -> str:
+        """A user-friendly name for the display type."""
+        return {
+            self.embedded_screen: "Embedded",
+            self.detailed_screen: "Detailed",
+            self.engineering_screen: "Engineering",
+        }[self]
```

Comment:
It's wild to me that this works, the code reads pretty confusingly if you forget that this is an `IntEnum` (as I did, on initial review).

I don't think many other classes would be able to look themselves up in a dictionary of their own attributes.

Reply:
I've used this pattern before with enums and kinda liked it - but if you think it's confusing it's worth redoing honestly as "clever code" is not good code

Reply:
I don't necessarily think it's worth changes- the only reason I was confused is because:

- The diff cut off at a point where I didn't see "IntEnum"
- I briefly forgot that IntEnum can be used as an int